Refactor: update main.js and  add class todoitem

Akramsous opened the pull request.

Akramsous committed:
feat(add,delete,filter,display):new fetures are added so now we are aple to add remove and display filtered tasks depending on it's priorities

Akramsous committed:
chore(structure): move css/js out of assets and update background color

Akramsous committed:
Merge branch 'main' into develop2

Akramsous committed:
Merge branch 'main' into develop2

Akramsous committed:
Merge branch 'develop2' of https://github.com/Akramsous/to-do-list into develop2

Akramsous committed:
todoitem : it's a class holds the the task properities

Akramsous committed:
refactor : inside main.js addTask function refactored with arrow function ,displayTasks refactored with arrow function and use create element and append child to display tasks also todoitem.js is added which contains ToDoItem class that describes task properity

Akramsous committed:
refactor: use todoitem.js instead of TodoItem.js

Akramsous committed:
refactor : update src of the js script in index.html and change displayTasks position in main.js file

Akramsous committed:
refactor : added todolist file contains ToDoList class to hold the tasks information and the functions it needs ,also updated main.js and removed uneccessary functions that could be in todolist.js ,finally added the nedded setter and getter to reach tasks information

oalfroukh commented on main.js:
```diff
@@ -0,0 +1,86 @@
+const addBtn = document.querySelector(".add-task-button");
+const taskList = document.getElementById("task-list");
+const filterPriority = document.getElementById("filter-priority");
+let toDoList = new ToDoList();
+
```
`const` and not `let`

oalfroukh commented on main.js:
```diff
@@ -0,0 +1,86 @@
+  const taskId = li.id;
+  const task = toDoList.getTasks().find((t) => t.getId() == taskId);
+
+  if (target.classList.contains("task-checkbox") || target.tagName === "SPAN"||) {
```
what does `target.tagName === "SPAN"||` mean?

oalfroukh commented on main.js:
```diff
@@ -0,0 +1,86 @@
+const filterPriority = document.getElementById("filter-priority");
+let toDoList = new ToDoList();
+
+const displayTasks = (filter = "all") => {
+  const tasks = toDoList.getTasks();
```
use `filterPriority` directly as it's general

oalfroukh commented on main.js:
```diff
@@ -0,0 +1,86 @@
+    const editBtn = document.createElement("button");
+    editBtn.className = "edit-task";
+    editBtn.textContent = "Edit";
+    if (task.isCompleted()) editBtn.disabled = true;
+
```
change the coclor of the button in case it's disabled

oalfroukh commented on main.js:
```diff
@@ -0,0 +1,86 @@
+    displayTasks(filterPriority.value);
+  }
+
+  if (target.classList.contains("delete")) {
+    toDoList.deleteTask(taskId);
+    displayTasks();
+  }
+  if (target.classList.contains("edit-task")) {
```
please use `addEventListener` for each button or checkbox and not depending on class name to do actions

oalfroukh commented on todoitem.js:
```diff
@@ -0,0 +1,43 @@
+    this.#priority = priority;
+    this.#completed = completed;
+  }
+  getId() {
+    return this.#id;
+  }
+  getName() {
+    return this.#name;
+  }
+  getPriority() {
+    return this.#priority;
+  }
+  isCompleted() {
+    return this.#completed;
+  }
+  setName(name) {
+    this.#name = name;
+  }
+  setPriority(priority) {
+    this.#priority = priority;
+  }
+
+  setCompleted(completed) {
+    this.#completed = completed;
+  }
+  toJSON() {
+    return {
+      id: this.#id,
+      name: this.#name,
+      priority: this.#priority,
+      completed: this.#completed,
+    };
+  }
```
delete teh not needed ones

Akramsous committed:
refactor : added the edit feature for tasks

Akramsous committed:
refactor: add event listneres to the delete and edit button and style the completed task edit buuton with another color also making todolis decleration const

Akramsous committed:
refactor : changing delete,edit and cancel implementation , moved inside displayTask finction and solving empty field problem

Akramsous committed:
fix: prevent 'Please fill out this field' popup when editing tasks

- Temporarily remove required attributes on task input and priority select during edit
- Restore required when saving or canceling the edit
- Prevent default form submission to allow editing without validation errors

Akramsous committed:
fix : save changes on the local storage after editing the task information

Akramsous committed:
refactor: seperate ui code from logices ,so moved the logical methods in to do list class , also changed the way of editing by use contenteditable feature

Akramsous committed:
refactor : removed uneccessary use of trim method in add task handling

Akramsous committed:
refactor : put all list buttons in one div

Akramsous committed:
fix : fixing when clicking edit on multiple list items then cancel or save it adds and remove d-none class name to all save and cancel buttons

Akramsous committed:
refactor : abstract the creation of elements

oalfroukh reviewed (approved):
Great implementation!
